govuk-one-login/di-account-management-frontend
# OLH-1721 Add error logging when we don't have a translation for content

## Proposed changes



### What changed

Added an error log for missing Welsh translation. Fallback language had to be removed from the initial i18next configuration in order to explicitly throw an error and still return the English translation. There are other alternatives like using `missingKeyHandler` option but it would not allow us to do what we wanted so a safeTranslation function was added in instead.

### Why did it change

To find in our logs which translation keys we are missing

### Related links




## Checklists


### Environment variables or secrets

## Testing

Unit tests
Local testing

## How to review

- Change a key in `cy/translation.json` e.g.  pages.security.header to pages.security.headerX
- Run the application locally and go to http://localhost:6001/security?lng=cy
- The error message should be displayed in the console as  ERROR (di-account-management-frontend/32301): TranslationError: key 'pages.security.header' missing for requested language.
